src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Equality.cs

```diff
@@ -149,218 +142,283 @@ public static bool Equals(ReadOnlySpan<byte> left, ReadOnlySpan<char> right)
+
+                        if (Vector256.GreaterThanAny((leftValues - vecA) & notEquals, vecZMinusA) || leftValues != rightValues)
+                        {
+                            return false; // first input isn't in [A-Za-z], and not exact match of lowered
+                        }
                     }
                 }
             }
 
             return true;
         }
 
-        /// <summary>
-        /// Determines whether the provided buffers contain equal ASCII characters, ignoring case considerations.
-        /// </summary>
-        /// <param name="left">The buffer to compare with <paramref name="right" />.</param>
-        /// <param name="right">The buffer to compare with <paramref name="left" />.</param>
-        /// <returns><see langword="true" /> if the corresponding elements in <paramref name="left" /> and <paramref name="right" /> were equal ignoring case considerations and ASCII. <see langword="false" /> otherwise.</returns>
-        /// <remarks>If both buffers contain equal, but non-ASCII characters, the method returns <see langword="false" />.</remarks>
-        public static bool EqualsIgnoreCase(ReadOnlySpan<byte> left, ReadOnlySpan<byte> right)
-            => left.Length == right.Length && SequenceEqualIgnoreCase(left, right);
-
-        /// <inheritdoc cref="EqualsIgnoreCase(ReadOnlySpan{byte}, ReadOnlySpan{byte})"/>
-        public static bool EqualsIgnoreCase(ReadOnlySpan<byte> left, ReadOnlySpan<char> right)
-            => left.Length == right.Length && SequenceEqualIgnoreCase(right, left);
-
-        /// <inheritdoc cref="EqualsIgnoreCase(ReadOnlySpan{byte}, ReadOnlySpan{byte})"/>
-        public static bool EqualsIgnoreCase(ReadOnlySpan<char> left, ReadOnlySpan<byte> right)
-            => EqualsIgnoreCase(right, left);
-
-        /// <inheritdoc cref="EqualsIgnoreCase(ReadOnlySpan{byte}, ReadOnlySpan{byte})"/>
-        public static bool EqualsIgnoreCase(ReadOnlySpan<char> left, ReadOnlySpan<char> right)
-            => left.Length == right.Length && SequenceEqualIgnoreCase(right, left);
-
-        private static bool SequenceEqualIgnoreCase<TLeft, TRight>(ReadOnlySpan<TLeft> left, ReadOnlySpan<TRight> right)
+        private interface ILoader<TLeft, TRight>
             where TLeft : unmanaged, INumberBase<TLeft>
             where TRight : unmanaged, INumberBase<TRight>
         {
-            Debug.Assert(left.Length == right.Length);
+            static abstract nuint Count128 { get; }
+            static abstract nuint Count256 { get; }
+            static abstract Vector128<TRight> Load128(ref TLeft ptr);
+            static abstract Vector256<TRight> Load256(ref TLeft ptr);
+        }
 
-            for (int i = 0; i < left.Length; i++)
-            {
-                uint valueA = uint.CreateTruncating(left[i]);
-                uint valueB = uint.CreateTruncating(right[i]);
+        private struct PlainLoader<T> : ILoader<T, T> where T : unmanaged, INumberBase<T>
```

Comment:
minor nit: it could be readonly?

```suggestion
        private readonly struct PlainLoader<T> : ILoader<T, T> where T : unmanaged, INumberBase<T>
```

Reply:
Yes it could. There are only `static` methods, thus I omited the `readonly` modifier.
Do we have any preference here? Some other structs, that implement static abstract interfaces, which I found in a quick search use it, so to be consistent it should be applied.